src/Microsoft.Diagnostics.Runtime/src/DacInterface/SosDac.cs

```diff
@@ -462,6 +469,32 @@ private string GetString(DacGetCharArrayWithArg func, ulong addr, bool skipNull
             return result;
         }
 
+        private string GetAsciiString(DacGetByteArrayWithArg func, ulong addr, bool skipNull = true)
+        {
+            int hr = func(Self, addr, 0, null, out int needed);
+            if (hr != S_OK)
+                return null;
+
+            if (needed == 0)
+                return "";
+            
+            byte[] buffer = AcquireBuffer(needed);
+            hr = func(Self, addr, needed, buffer, out needed);
+            if (hr != S_OK)
+            {
+                ReleaseBuffer(buffer);
+                return null;
+            }
+
+            if (skipNull)
```

Comment:
Maybe we could remove skipNull and be more permissive with what we accept?  I'm thinking something like:

    int len = Array.IndexOf(buffer, 0);
    if (len == -1)
        len = needed;
    string result = Encoding.ASCII.GetString(buffer, 0, len);

It's _slightly_ less efficient than assuming we know if the string will be null terminated or not, but the dac is a really fickle library (meaning, very poorly implemented).  I prefer to not assume that strings will/won't be properly null terminated.

Reply:
I've removed the `skipNull` arg. The method is a copy of the other GetString() method which reads UTF-16 data. If needed, the methods could be merged but would IMO be a little ugly with an extra 'isUnicode' or similar parameter.

Reply:
I like having two separate methods.